migrate_cpp/cpp_refactoring/rewriter_test.cpp

```diff
@@ -0,0 +1,133 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "migrate_cpp/cpp_refactoring/rewriter.h"
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "clang/AST/ASTConsumer.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendAction.h"
+#include "clang/Tooling/Tooling.h"
+
+namespace Carbon::Testing {
+namespace {
+
+// Represents C++ source code with at most one region enclosed in $[[...]]$ as
+// an annotated range.
+class Annotations {
+ public:
+  Annotations(std::string_view annotated_source) {
+    size_t index = annotated_source.find("$[[");
+    if (index == std::string_view::npos) {
+      source_code = std::string(annotated_source);
+      return;
+    }
+    start = index;
+    end = annotated_source.find("]]$", index);
```

Comment:
```suggestion
    end = annotated_source.find("]]$", index);
    CARBON_CHECK(end != std::string_view::npos) << "Found $[[, but missing ]]$.";
```

Suggesting as a check because this would be simple developer error in the test string, but I think it's best to validate.

Reply:
Done.